### PR TITLE
Adding coverage reporter

### PR DESCRIPTION
**Pull request by citizenken**
This publisher adds support for [this plugin](https://github.com/jenkinsci/github-pr-coverage-status-plugin).

### CODE REVIEW

File: spec/lib/jenkins_pipeline_builder/extensions/wrappers_spec.rb [hunk 1]
```diff
@@ -39,6 +39,7 @@
     end
 
     it 'fails parameters are passed' do
+      pending
```
Comment (JustMcAfee) [inline spec/lib/jenkins_pipeline_builder/extensions/wrappers_spec.rb]: Why is this pending?

File: lib/jenkins_pipeline_builder/extensions/publishers.rb [hunk 2]
```diff
@@ -682,3 +676,21 @@
...
+publisher do
+  name :github_pr_coverage_status_reporter
+  plugin_id 'github-pr-coverage-status'
+  description 'Code coverage icon for GitHub pull requests'
+  jenkins_name 'GitHub Pull Request Coverage Status'
+  announced false
+
+  xml do |action|
+    if action == 'compare'
+      send('com.github.terma.jenkins.githubprcoveragestatus.CompareCoverageAction')
```
Comment (JustMcAfee) [inline lib/jenkins_pipeline_builder/extensions/publishers.rb]: Is this it?  Just one XML tag needed?

Comment (citizenken) [reply lib/jenkins_pipeline_builder/extensions/publishers.rb]: Yep, there is only 1 XML tag needed, as the configuration is at the global level

File: spec/lib/jenkins_pipeline_builder/extensions/wrappers_spec.rb (hunk 1 above)
Comment (citizenken) [reply spec/lib/jenkins_pipeline_builder/extensions/wrappers_spec.rb]: I have added a comment for more context on why this is pending

**Pull request merged**